pulsar-client-cpp/lib/ClientImpl.cc

```diff
@@ -129,7 +129,7 @@ ClientImpl::ClientImpl(const std::string& serviceUrl, const ClientConfiguration&
     }
 }
 
-ClientImpl::~ClientImpl() { shutdown(); }
```

Comment:
Wouldn't this leave client running if shutdown is not called explicitly?

Reply:
It will cause memory leak that the internal producers, consumers, `ClientConnection` and `ExecutorService` are not closed. I'll fix it,

Reply:
I think there's no need to call `shutdown()` in destructor. `shutdown()` is an exposed method that is responsible to release all resources and can be called by `close()`. User should explicitly call `close()` or `shutdown()` otherwise there will be a resource leak.

Reply:
That would be a breaking change in the semantic. Right now the destructor is taking care of automatically releasing all the resources

Reply:
Got it. I'll restore this behavior.